src/interfaces/ffi/api.rs

```diff
@@ -36,80 +37,17 @@ use crate::{
     },
 };
 
-/// Search
-///  - `base64_keywords`        : an array of base64 encoded `Keyword` bytes
-///  - `master_key_bytes`       : the bytes of the serialized master key
-///  - `label_bytes`            : additional information used to derive Entry
-///    Table UIDs
-///  - `max_results_per_keyword`: maximum number of results to return per
-///    keyword
-/// - `max_depth`               : maximum recursion depth allowed
-/// - `progress_callback`       : callback used to retrieve intermediate results
-///   and transmit user interrupt
-///  - `fetch_entry`            : the callback function to fetch the values from
-///    the Entry Table
-///  - `fetch_chain`            : the callback function to fetch the values from
-///    the Chain Table
-#[allow(clippy::too_many_arguments)]
-async fn ffi_search(
-    base64_keywords: &[String],
-    master_key_bytes: &[u8],
-    label_bytes: &[u8],
-    max_results_per_keyword: usize,
-    max_depth: usize,
-    fetch_chains_batch_size: NonZeroUsize,
-    progress: ProgressCallback,
-    fetch_entry: FetchEntryTableCallback,
-    fetch_chain: FetchChainTableCallback,
-) -> Result<Vec<u8>, FindexErr> {
-    let master_key = KeyingMaterial::<MASTER_KEY_LENGTH>::try_from_bytes(master_key_bytes)
-        .map_err(|e| {
-            FindexErr::Other(format!("While parsing master key for Findex search, {e}"))
-        })?;
-    let label = Label::from(label_bytes);
-
-    let mut keywords = HashSet::with_capacity(base64_keywords.len());
-    for base64_keyword in base64_keywords {
-        // base64 decode the words
-        let word_bytes = base64::decode(base64_keyword).map_err(|e| {
-            FindexErr::ConversionError(format!(
-                "Failed decoding the base64 encoded word: {base64_keyword}: {e}"
-            ))
-        })?;
-        keywords.insert(Keyword::from(word_bytes));
-    }
-
-    let mut ffi_search = FindexUser {
-        progress: Some(progress),
-        fetch_entry: Some(fetch_entry),
-        fetch_chain: Some(fetch_chain),
-        upsert_entry: None,
-        insert_chain: None,
-        update_lines: None,
-        list_removed_locations: None,
-        fetch_all_entry_table_uids: None,
-    };
-
-    let res = ffi_search
-        .search(
-            &keywords,
-            &master_key,
-            &label,
-            max_results_per_keyword,
-            max_depth,
-            fetch_chains_batch_size,
-            0,
-        )
-        .await?;
-
-    // Serialize the results.
-    let mut serializer = Serializer::new();
-    serializer.write_u64(res.len() as u64)?;
-    for (keyword, locations) in res {
-        serializer.write_vec(&keyword)?;
-        serializer.write(&SerializableSet(&locations))?;
-    }
-    Ok(serializer.finalize())
+/// Re-export the `cosmian_ffi` `h_get_error` function to clients with the old
```

Comment:
we need to configure cbindgen.toml as we did for CoverCrypt (to have cosmian_ffi functions also exported in the findex.h)

Reply:
Ok todo in another PR

Reply:
I am planning to merged `cosmian_ffi` to `cloudproof_rust`. Everything will be in one crate, things will be simpler :)